releasenotes/notes/parallel-check-public-7faed5f3e20e1d03.yaml

```diff
@@ -0,0 +1,7 @@
+---
+features_misc:
+  - |
+    A new function, :func:`qiskit.utils.should_run_in_parallel`, is available to determine whether
+    :func:`.parallel_map` will choose to run in parallel or degrade to running in serial.  This
+    decision is dependent on how many CPUs are available to Qiskit, what the :mod:`multiprocessing`
+    start method is, how many processes were requested.
```

Comment:
I mean sort of, we don't explicitly check what multiprocessing is set to (ie we ignore if the user explicitly sets this to spawn), but the value for `PARALLEL_DEFAULT` is based on what the OS default start method is.

Reply:
Yeah, I think I was being more aspirational than correct here. We _should_ do the check based on the `multiprocessing` start method if that's what we care about, but we don't. I can change the wording.

Reply:
Maybe we should just do this, it seems totally in scope to update the logic in the function to call ` multiprocessing.get_start_method()` and get rid of the OS based logic here for 1.2.

Reply:
> Maybe we should just do this, it seems totally in scope to update the logic in the function to call multiprocessing.get_start_method() and get rid of the OS based logic here for 1.2.

@mtreinish is this still something we ought to do prior to merging this for 1.3?

Reply:
Yeah, we still intended to do this